lib/shoryuken/polling/strict_priority.rb

```diff
@@ -73,11 +78,22 @@ def queue_paused?(queue)
         @paused_until[queue] > Time.now
       end
 
-      def pause(queue)
+      def delay_pause(queue)
         return unless delay > 0
 
         @paused_until[queue] = Time.now + delay
-        logger.debug "Paused #{queue}"
+        logger.debug "Paused #{queue}, no messages found"
+      end
+
+      def interval_pause(queue, messages_found)
+        return unless interval > 0
+
+        @paused_until[queue] = Time.now + interval * messages_found
```

Comment:
I was worried that shoryuken would be pulling lots of messages off the queue at once and this would be a large number, but from googling it looks like they only pull a max of 10 messages at a time. Is that your understanding as well? If that's the case I think this will work well.

Reply:
It pulls up to `concurrency` number of messages, which is a value we set in the configuration value. For our use-case this isn't relevant though since FIFO queues are maxed out at 1.

Reply:
Got it, sounds good.